Modules/_sqlite/connection.c

```diff
@@ -1648,60 +1646,51 @@ pysqlite_connection_backup_impl(pysqlite_Connection *self,
     bck_handle = sqlite3_backup_init(bck_conn, "main", self->db, name);
     Py_END_ALLOW_THREADS
 
-    if (bck_handle) {
-        do {
-            Py_BEGIN_ALLOW_THREADS
-            rc = sqlite3_backup_step(bck_handle, pages);
-            Py_END_ALLOW_THREADS
+    if (bck_handle == NULL) {
+        _pysqlite_seterror(bck_conn, NULL);
+        return NULL;
+    }
 
-            if (progress != Py_None) {
-                PyObject *res;
-
-                res = PyObject_CallFunction(progress, "iii", rc,
-                                            sqlite3_backup_remaining(bck_handle),
-                                            sqlite3_backup_pagecount(bck_handle));
-                if (res == NULL) {
-                    /* User's callback raised an error: interrupt the loop and
-                       propagate it. */
-                    callback_error = 1;
-                    rc = -1;
-                } else {
-                    Py_DECREF(res);
-                }
-            }
+    int rc;
```

Comment:
Moving definition of rc here is not really necessary.

Reply:
Correct. I moved it because IMO readability is improved when the variables declarations are close to where they are used. I'll revert it.